### ISSUE TITLE
Setting `is_adapted` to True when PreprocessingLayer is initialized with parameters

### ISSUE THREAD

**Issue by szleb** [keras-team/keras#16415]
**System information**.

TensorFlow version (you are using): 2.8.0
Are you willing to contribute it (Yes/No) : No

**Feature and the current behavior/state**.

When a PrerpocessingLayer is adapted the attribute `is_adapted` is set to True. Calling `adapt` a second time (possibly with different data) updates the parameters of the PreprocessingLayer to the new passed data (at least in some cases, e.g. with tf.keras.layers.Normalization, but does not work with tf.keras.layers.StringLookups). The user, however, can check if the PreprocessingLayer has been adapted already by looking at the attribute `is_adapted`.

When the PreprocessingLayer is instantiated in a way that all parameters are set, e.g. tf.keras.layers.Normalization with mean and variance, calling `adapt` will raise an error. But the attribute `is_adapted` is not set to `True`, so the user cannot check before calling `adapt` if the layer has been (manually) adapted already.

It would be very helpful, when the attribte `is_adapted` is set to `True` when the PreprocessingLayer has been provided with the necessary parameters when being initialized.

**Comment by tilakrayal** [comment]
Hello @szleb ,
I request you please specify any Use Cases for this feature where we can use the mentioned feature. Thanks!

**Comment by szleb** [comment]
**Use Case**

In my use case, I get user defined preprocessing layers. When I build the model I need to know if the parameters of the layers has been set manually or not. In the first scenario, I can use the layer as is and in the second scenario I need to call adapt.  If I just call `adapt` without knowing if necessary, an error is raised in the first scenario. So currently I just do the following:
   
    try:
        preprocessing_layer.adapt(data)
    except ValueError:
        pass

This is not very convenient, as other causes raising a Value Error might be missed. With the suggested feature the attribute `is_adapted` of the preprocessing layer could be checked :

    if not preprocessing_layer.is_adapted:
        preprocessing_layer.adapt(data)
